beacon-chain/core/peerdas/helpers.go

```diff
@@ -205,39 +206,29 @@ func DataColumnSidecarsForReconstruct(
 	blobKzgCommitments [][]byte,
 	signedBlockHeader *ethpb.SignedBeaconBlockHeader,
 	kzgCommitmentsInclusionProof [][]byte,
-	blobs []kzg.Blob,
+	cellsAndProofs []kzg.CellsAndProofs,
 ) ([]*ethpb.DataColumnSidecar, error) {
-	blobsCount := len(blobs)
+	// Each CellsAndProofs corresponds to a Blob
+	// So we can get the BlobCount by checking the length of CellsAndProofs
+	blobsCount := len(cellsAndProofs)
 	if blobsCount == 0 {
 		return nil, nil
 	}
 
-	// Compute cells and proofs.
-	cells := make([][kzg.CellsPerExtBlob]kzg.Cell, 0, blobsCount)
-	proofs := make([][kzg.CellsPerExtBlob]kzg.Proof, 0, blobsCount)
-
-	for i := range blobs {
-		blob := &blobs[i]
-		blobCells, blobProofs, err := kzg.ComputeCellsAndKZGProofs(blob)
-		if err != nil {
-			return nil, errors.Wrap(err, "compute cells and KZG proofs")
-		}
-
-		cells = append(cells, blobCells)
-		proofs = append(proofs, blobProofs)
-	}
-
 	// Get the column sidecars.
 	sidecars := make([]*ethpb.DataColumnSidecar, 0, kzg.CellsPerExtBlob)
 	for columnIndex := uint64(0); columnIndex < kzg.CellsPerExtBlob; columnIndex++ {
 		column := make([]kzg.Cell, 0, blobsCount)
 		kzgProofOfColumn := make([]kzg.Proof, 0, blobsCount)
 
 		for rowIndex := 0; rowIndex < blobsCount; rowIndex++ {
-			cell := cells[rowIndex][columnIndex]
+			cellsForRow := cellsAndProofs[rowIndex].Cells
+			proofsForRow := cellsAndProofs[rowIndex].Proofs
+
+			cell := cellsForRow[columnIndex]
 			column = append(column, cell)
 
-			kzgProof := proofs[rowIndex][columnIndex]
+			kzgProof := proofsForRow[columnIndex]
 			kzgProofOfColumn = append(kzgProofOfColumn, kzgProof)
 		}
```

Comment:
Just for my understanding; this code is iterating over each rowIndex(blobIndex) and getting the corresponding cells and proofs for that Blob. The code is then using the column index to fetch the cell and proof for that blob that corresponds to the column that we want